src/sentry/replays/post_process.py

```diff
@@ -2,23 +2,87 @@
 
 import collections
 from itertools import zip_longest
-from typing import Any, Generator, Iterable, Iterator, MutableMapping
-
-
-def process_raw_response(response: list[dict[str, Any]], fields: list[str]) -> list[dict[str, Any]]:
+from typing import Any, Dict, Generator, Iterable, Iterator, List, MutableMapping, Optional, Union
+
+from typing_extensions import TypedDict
+
+
+class DeviceResponseType(TypedDict, total=False):
+    name: Optional[str]
+    brand: Optional[str]
+    model: Optional[str]
+    family: Optional[str]
+
+
+class SDKResponseType(TypedDict, total=False):
+    name: Optional[str]
+    version: Optional[str]
+
+
+class OSResponseType(TypedDict, total=False):
+    name: Optional[str]
+    version: Optional[str]
+
+
+class BrowserResponseType(TypedDict, total=False):
+    name: Optional[str]
+    version: Optional[str]
+
+
+class UserResponseType(TypedDict, total=False):
+    id: Optional[str]
+    username: Optional[str]
+    email: Optional[str]
+    ip: Optional[str]
+    display_name: Optional[str]
+
+
+class ReplayDetailsResponse(TypedDict, total=False):
+    id: str
+    project_id: str
+    trace_ids: List[str]
+    error_ids: List[str]
+    environment: Optional[str]
+    tags: Union[Dict[str, List[str]], List]
+    user: UserResponseType
+    sdk: SDKResponseType
+    os: OSResponseType
+    browser: BrowserResponseType
+    device: DeviceResponseType
+    is_archived: Optional[bool]
+    urls: Optional[List[str]]
+    clicks: List[Dict[str, Any]]
```

Comment:
i wonder if we could type the clicks object?

Reply:
discussed offline